### PR TITLE
fix PV watcher on secret - in case of topology

### PR DESCRIPTION
**Pull request by alonfirestein**

### CODE REVIEW

File: controllers/util/poolutil.go [hunk 1]
```diff
@@ -129,6 +141,44 @@ var SecretMgmtAddrPredicate = predicate.Funcs{
...
+	err = json.Unmarshal(newSecret.Data[TopologySecretDataKey], &newSecretMgmtDataByMgmtId)
+	if err != nil {
+		return false
+	}
+	for mgmtId := range oldSecretMgmtDataByMgmtId {
+		if _, exist := newSecretMgmtDataByMgmtId[mgmtId]; exist {
+			oldSecretMgmtAddr := oldSecretMgmtDataByMgmtId[mgmtId][SecretManagementAddressKey]
+			newSecretMgmtAddr := newSecretMgmtDataByMgmtId[mgmtId][SecretManagementAddressKey]
+			if oldSecretMgmtAddr != newSecretMgmtAddr {
+				return false
```
Comment (tyichye19) [inline controllers/util/poolutil.go:174]: should return true

File: controllers/util/poolutil.go [hunk 2]
```diff
@@ -129,6 +141,44 @@ var SecretMgmtAddrPredicate = predicate.Funcs{
...
+			}
+		}
+		return false
```
Comment (tyichye19) [inline controllers/util/poolutil.go:177]: should return true

**Review by tyichye19 — APPROVED** [review]
LGTM

**Comment by bvered** [comment]
[PV fix watcher on secret - in case of topology](https://trello.com/c/279T5YpX/227-pv-fix-watcher-on-secret-in-case-of-topology)

**Pull request merged**